packages/twenty-front/src/modules/ui/navigation/navigation-drawer/components/NavigationDrawerItem.tsx

```diff
@@ -185,6 +187,7 @@ export const NavigationDrawerItem = ({
         as={to ? Link : 'div'}
```

Comment:
Setting `role=button` is half a solution to make that component a button. Instead, we should render a `<button>` HTML tag when we actually want to display a button. The `as` property allows to choose which component we want to render in the end. We should use it to display a button when we want.

Settings the `role` attribute of an HTML element is last resort. We should always try to use semantic HTML at first, and here using a `button` element makes sense.

To make it work with the current code, the `NavigationDrawerItem` component could take an optional `as` property which could be `'button' | undefined`. We could use it to choose which component to render.